app/services/get_user_coordinates_service.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class GetUserCoordinatesService < BaseService
+  LIMIT_LENGTH = 256
+
+  def initialize(params:)
+    super()
+    @ward = params[:ward]
+    @district = params[:district]
+  end
+
+  def call
+    validate!
+    return self if error?
+
+    fetch_request
+  end
+
+  private
+
+  def validate!
+    return add_error(I18n.t('errors.params.empty', param: :address_search)) if address_search.blank?
+
+    add_error(I18n.t('errors.params.too_long', param: :address_search)) if address_search.length > LIMIT_LENGTH
```

Comment:
this one may occur in other user search as well, try to handle it in global, or we have to duplicated it in other places

Reply:
Noted, there will be another PR to handle this case